File: spring-2018/test_1/4_fibonacci.m
clc
clear

function is =isPerfectSquare(x)
    s = fix(sqrt(x));
    is= (s*s == x);
end

function s=isFibonacci(n)
    s=isPerfectSquare(5*n*n + 4) || isPerfectSquare(5*n*n - 4);
end

% for i=1:1000
%     fibotest=isFibonacci(i);

n=input('')
fibotest=isFibonacci(n);

if (fibotest==1)
    fprintf('Yes');
else
    fprintf('No');
end

File: spring-2018/test_1/4_fibonacci.py
from random import randint
import math

RAND_RANGE = (1, 50)

####################################

NUM_CASES = 20

TEST_CASE_FORMAT = """
case = Test %d
input = %d
output = /.*%s.*/
"""

####################################


def is_perf_square(n):
    s = math.floor(math.sqrt(n))
    return s * s == n


def is_fib(n):
    return is_perf_square(5 * n * n + 4) or is_perf_square(5 * n * n - 4)


for i in range(1, 1 + NUM_CASES):

    inp = randint(*RAND_RANGE)
    out = "Yes" if is_fib(inp) else "No"

    print(TEST_CASE_FORMAT % (i, inp, out))
